### ISSUE TITLE
When page parameter passed is greater than 999 (max pages), the search interface should not show actual results total

### ISSUE THREAD

**Issue by Justin-Pickett** [department-of-veterans-affairs/va.gov-team#2268]
@kfrz commented on [Fri Nov 02 2018](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/14842)

## Background
When a user navigates to a search page that may have greater than 999 results, and manually ticks the URL `page` param to larger integer than 99, then they are presented with 

> Showing 14281-14290 of 999 results

This is confusing from a UX standpoint.

## Screenshots
[Normal page of results](https://staging.va.gov/search/?page=2&query=test)
![2018-11-02-160007_321x247_scrot.png](https://images.zenhubusercontent.com/5951581361b4ca3b0a725a57/6fe4e416-77ac-4a5d-8c62-b908dd949132)

[Hacker enters a larger page number](https://staging.va.gov/search/?page=100&query=test)
![2018-11-02-160207_566x259_scrot.png](https://images.zenhubusercontent.com/5951581361b4ca3b0a725a57/8f2fce98-779e-4d9b-8141-9ceaba9b5f2a)

## Acceptance Criteria
- [ ] When a user navigates to a search results page with a page param that exceeds the range of visible pages, they should see the same "totals" copy in the interface and be directed to the 99th (or last) page. 

---

@batemapf commented on [Mon Sep 23 2019](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/14842#issuecomment-534148911)

Doing some repo cleanup. @chrisj-usds @Karak888 I believe this is one for Global UX. If so, would you please assign yourselves and, if you are migrating issues to `va.gov-team`, please take care of this one?

**Comment by chrisj-usds** [comment]
@Justin-Pickett Adding the global-ux team as on-site search is in their purview. Thanks for flagging!

**Comment by Justin-Pickett** [comment]
@chrisj-usds, Sure thing. Thank you!

**Comment by Karak888** [comment]
yea @chrisj-usds , thanks a lot! ;)

**Comment by Karak888** [comment]
will leave this until @kelsonic and @ncksllvn are back and available to size and work this ticket.

**Comment by Karak888** [comment]
needs to be validated.

**Comment by ncksllvn** [comment]
Update on this - this unexpectedly appears to be an API issue. When the page loads, the front-end sends a request the API at `https://api.va.gov/v0/search?query=test&page=100`. It then responds with this data -

![image.png](https://images.zenhubusercontent.com/59cb9de1b0222d5de47953d8/dde1091d-f1f4-4e76-8c2b-7e6344173b0e)

This is what we see rendered on the front-end, so seems like the pagination limit is 99 pages before we start seeing unexpected responses like we do in this ticket. I'm thinking of what the best fix would be.

**Comment by kelsonic** [comment]
I think if `totalPages` remains the same, that's fine for our current UI/UX. But perhaps we can update `totalEntries` on the API to reflect the actual total amount of entries!

**Comment by ncksllvn** [comment]
Funny, there's a [comment in Vets-API](https://github.com/department-of-veterans-affairs/vets-api/blob/b1f7613b5aa568cdb81c5f76223cf18479d13883/lib/search/pagination.rb#L17) that describes this issue -

```
# Due to Search.gov's offset max of 999, we cannot view pages
    # where the offset param exceeds 999.  This influences our:
    #   - total_viewable_pages
    #   - total_viewable_entries
    #
    # @see https://search.usa.gov/sites/7378/api_instructions under `offset`
    #
```

Since we're on page 100, and there are 10 results per page, I think we're running into the mysteriousness of that limit on the offset max. I also notice weirdness on page 99.

https://www.va.gov/search/?page=99&query=test

There are 10 results per page, but the results say -

![image](https://user-images.githubusercontent.com/1915775/69566155-71595100-0f84-11ea-966b-4fae02b73e21.png)

**Comment by ncksllvn** [comment]
So I guess the proper behavior of this app would be to show an error if we manually navigate past page 99, or to validate the page number by defaulting to 99 if greater.

**Comment by kfrz** [comment]
Ah yes, now I am recalling implementation from many months ago... implementation which I did. Just a cursory glance over their docs seems nothing has changed. Happy to pitch in on the API changes if need be.

**Comment by ncksllvn** [comment]
Thanks @kfrz! What would you advise here? Throw an error from the API if the pagination limit is exceeded?

**Comment by ncksllvn** [comment]
PR is open for this. We may need a follow-up though. I think page 100 (result 991-1000) should be accessible, but it isn't right now.

**Comment by ncksllvn** [comment]
Merged, and should be on Staging in a half hour.

**Comment by Karak888** [comment]
moving to close as PR is merged.  Thanks @ncksllvn - if there is something additional needed as a follow up let's create a new ticket.  but closing this for now!
